# Alessandro's Data Analysis and Exploration on MCF7 SMARTS

In [1]:
#initial imports
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import kurtosis, skew
import sklearn.feature_selection as fs
from sklearn.preprocessing import StandardScaler
from sklearn.decomposition import PCA

In [2]:
unfiltered_mcf7_smarts = pd.read_csv("data/smartSeq/MCF7_SmartS_Unfiltered_Data.txt", sep = " ")
meta__mcf7_smarts = pd.read_table("data/smartSeq/MCF7_SmartS_MetaData.tsv")

In [7]:
unfiltered_mcf7_smarts.head()

output.STAR.1_A10_Hypo_S28_Aligned.sortedByCoord.out.bam  \
WASH7P                                                     0          
MIR6859-1                                                  0          
WASH9P                                                     1          
OR4F29                                                     0          
MTND1P23                                                   0          

           output.STAR.1_A11_Hypo_S29_Aligned.sortedByCoord.out.bam  \
WASH7P                                                     0          
MIR6859-1                                                  0          
WASH9P                                                     0          
OR4F29                                                     0          
MTND1P23                                                   0          

           output.STAR.1_A12_Hypo_S30_Aligned.sortedByCoord.out.bam  \
WASH7P                                                     0          
MIR6859-1                                                  0          
WASH9P                                                     0          
OR4F29                                                     0          
MTND1P23                                                   0          

           output.STAR.1_A1_Norm_S1_Aligned.sortedByCoord.out.bam  \
WASH7P                                                     0        
MIR6859-1                                                  0        
WASH9P                                                     0        
OR4F29                                                     0        
MTND1P23                                                   0        

           output.STAR.1_A2_Norm_S2_Aligned.sortedByCoord.out.bam  \
WASH7P                                                     0        
MIR6859-1                                                  0        
WASH9P                                                     0        
OR4F29                                                     0        
MTND1P23                                                   0        

           output.STAR.1_A3_Norm_S3_Aligned.sortedByCoord.out.bam  \
WASH7P                                                     0        
MIR6859-1                                                  0        
WASH9P                                                     1        
OR4F29                                                     0        
MTND1P23                                                   0        

           output.STAR.1_A4_Norm_S4_Aligned.sortedByCoord.out.bam  \
WASH7P                                                     0        
MIR6859-1                                                  0        
WASH9P                                                    10        
OR4F29                                                     0        
MTND1P23                                                   0        

           output.STAR.1_A5_Norm_S5_Aligned.sortedByCoord.out.bam  \
WASH7P                                                     0        
MIR6859-1                                                  0        
WASH9P                                                     1        
OR4F29                                                     0        
MTND1P23                                                   0        

           output.STAR.1_A6_Norm_S6_Aligned.sortedByCoord.out.bam  \
WASH7P                                                     0        
MIR6859-1                                                  0        
WASH9P                                                     0        
OR4F29                                                     0        
MTND1P23                                                   0        

           output.STAR.1_A7_Hypo_S25_Aligned.sortedByCoord.out.bam  ...  \
WASH7P                                                     0        ...   
MIR6859-1                                                  0        ...   
WASH9P         

In the first part of data analysis we take a look at the dataset.



In [8]:
unfiltered_mcf7_smarts.shape

(22934, 383)

The columns of the MCF7 SmartSeq dataset represent the different cells from the cell line. They are 383 in total.

The rows of the MCF7 SmartSeq dataset represent the different genes present in each cell and in what quantity. They are 22934 in total.

## Cleaning the data

### Cheking for null entries

In [9]:
print(unfiltered_mcf7_smarts.isnull().sum())
print(f"\nColumns with null data: {unfiltered_mcf7_smarts.columns[unfiltered_mcf7_smarts.isnull().any()].to_list()}")

output.STAR.1_A10_Hypo_S28_Aligned.sortedByCoord.out.bam    0
output.STAR.1_A11_Hypo_S29_Aligned.sortedByCoord.out.bam    0
output.STAR.1_A12_Hypo_S30_Aligned.sortedByCoord.out.bam    0
output.STAR.1_A1_Norm_S1_Aligned.sortedByCoord.out.bam      0
output.STAR.1_A2_Norm_S2_Aligned.sortedByCoord.out.bam      0
                                                           ..
output.STAR.4_H5_Norm_S359_Aligned.sortedByCoord.out.bam    0
output.STAR.4_H6_Norm_S360_Aligned.sortedByCoord.out.bam    0
output.STAR.4_H7_Hypo_S379_Aligned.sortedByCoord.out.bam    0
output.STAR.4_H8_Hypo_S380_Aligned.sortedByCoord.out.bam    0
output.STAR.4_H9_Hypo_S381_Aligned.sortedByCoord.out.bam    0
Length: 383, dtype: int64

Columns with null data: []


So we can clearly see that there is no null data in the dataset.

### Checking for outliers